doc/vhpidirect/examples/arrays.rst

```diff
@@ -0,0 +1,56 @@
+.. program:: ghdl
+.. _COSIM:VHPIDIRECT:Examples:arrays:
+
+Arrays
+######
+
+.. _COSIM:VHPIDIRECT:Examples:arrays:logicvectors:
+
+:cosimtree:`logicvector <vhpidirect/arrays/logicvector>`
+**************************************************************
+
+Commonly signals in VHDL are of a logic type or a vector thereof (``std_logic`` and ``std_logic_vector``), coming from IEEE's ``std_logic_1164`` package.
+These types can hold values other than high and low (``1`` and ``0``) and are enumerated as:
+
+0. 'U' 
+1. 'X' 
+2. '0' 
+3. '1' 
+4. 'Z' 
+5. 'W' 
+6. 'L' 
+7. 'H' 
+8. '-' 
+
+Because the number of enumeration values is less than 256, logic values are transported in 8 bit words (a ``char`` type in C) (as mentioned in :ref:`Restrictions_on_foreign_declarations`).
```

Comment:
Since `Restrictions_on_foreign_declarations` is the reference for both sentences/assertions, I think we can make it more explicit:

As mentioned in Restriction...:

* Because the number...
* Logic vectors,...

Moreover, I would:

- Reorder them.
- Indent "In this example two declarations...." and the list below, to be part of the item about "enumeration values".

Then, after the list/clarification about the types and helpers that we use for std_logic, the next paragraph would follow, "This examples builds...", as it is now (unindented).

Reply:
I have left the original order (char for the enums, then char[] for the logic vectors) as it reads better than them flipped (if that is even what you were suggesting).

As mentioned in Restriction:
 - Because the number of...
   - In this example 2 declarations.
 - Logic vectors as char[]

This example builds

Reply:
I'm confused. "This example builds on the integer vector example" (and the HINT) should go before ""For illustrative purposes, the two vectors are.." (and the ATTENTION), isn't it? All other changes look good.

Reply:
Can do... I think I was getting confused..The [host](https://ghdl-cosim.readthedocs.io/en/logicvectors/vhpidirect/examples/arrays.html#logicvector) has updated.

Reply:
It's ok. My browser cache was also laughing at me... Both look good now, the sources and the site.